src/com/company/Shop1.java

```diff
@@ -1,88 +1,94 @@
+    private Map<Product, Integer> mapOfShoppingCart = new HashMap<>();
     private int intCounter;
-    private double doubleCounter;
-    public Shop1(Map check, double doubleCounter){
-        this.check = check;
-        this.doubleCounter =doubleCounter;
-    }
-    public Shop1(){
-    }
 
-    public List<Product> getProductList() {
-        return productList;
-    }
 
-    public void setProductList(List<Product> productList) {
-        this.productList = productList;
+    public Shop1() {
     }
 
-    public void setCheck(Map<String, Integer> check) {
-        this.check = check;
-    }
 
-    public int getIntCounter() {
-        return intCounter;
-    }
-
-    public void setIntCounter(int intCounter) {
-        this.intCounter = intCounter;
-    }
 
-    public double getDoubleCounter() {
-        return doubleCounter;
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Shop1 shop1 = (Shop1) o;
+        return Objects.equals(storageMap, shop1.storageMap);
     }
 
-    public void setDoubleCounter(double doubleCounter) {
-        this.doubleCounter = doubleCounter;
+    @Override
+    public int hashCode() {
+        return Objects.hash(storageMap);
     }
 
-    public String toString() {
 
-        return check+"\n"+Double.toString(doubleCounter);
-    }
-
-    public List addProduct(Product product){
+    public void addProduct(Product product) {
         productList.add(product);
-        return  productList;
-    }
+        intCounter = 0;
+
+        if (mapOfId.containsKey(product)) {
+            intCounter++;
+        } else {
+            intCounter = 1;
+        }
+        mapOfId.put(product.getId(), product);
 
 
-    public Shop1 getCheck(){ //Можно ли так делать? Сделал для того, что обшая сумма была после чека, а не до.
         for (int i = 0; i < productList.size(); i++) {
             for (int j = 0; j < productList.size(); j++) {
                 if (productList.get(i).getName().equals(productList.get(j).getName())) {
                     intCounter++;
                 }
             }
-            check.put(productList.get(i).getName(), intCounter);
+            storageMap.put(productList.get(i), intCounter);
+
 
             intCounter = 0;
         }
 
-        totalAmount();
-        Shop1 shop1 = new Shop1(check, doubleCounter);
-        return shop1;
 
+    }
+
+
+    public Paycheck getCheck(Buyer buyer) {
+
+        Paycheck paycheck = new Paycheck();
+
+        paycheck.put(buyer.getMapOfShoppingCart());
+
+        return paycheck;
 
 
     }
-    private double totalAmount(){
-        for (int i = 0; i < productList.size() ; i++) {
-            doubleCounter+= productList.get(i).getCost();
 
-        }
-        return  doubleCounter;
+    public Map<Product, Integer> getStorageMap() {
+        return storageMap;
     }
 
+    public Map<Product, Integer> giveProduct(long id) {
```

Comment:
what do you mean by give product?
your task was to create a Check
not to give any products